python/feathub/tests/feathub_it_test_base.py

```diff
@@ -106,6 +111,47 @@ def create_file_source(
             timestamp_format=timestamp_format,
         )
 
+    # TODO: only invoke the corresponding base class's setUpClass()
+    #  method to reduce resource consumption.
+    @classmethod
+    def invoke_all_base_class_setupclass(cls):
+        for base_class in cls.__bases__:
+            if issubclass(base_class, unittest.TestCase):
+                base_class.setUpClass()
+
+    @classmethod
+    def invoke_all_base_class_teardownclass(cls):
+        for base_class in cls.__bases__:
+            if issubclass(base_class, unittest.TestCase):
+                base_class.tearDownClass()
+
+    def invoke_base_class_setup(self):
+        self._get_base_test_class().setUp(self)
+
+    def invoke_base_class_teardown(self):
+        self._get_base_test_class().tearDown(self)
+
+    def _get_base_test_class(self) -> Type[unittest.TestCase]:
+        if self._base_class_mapping is None:
+            self._base_class_mapping = dict()
+            for base_class in self.__class__.__bases__:
+                if not issubclass(base_class, unittest.TestCase):
```

Comment:
nits: it might be more readable to write it in this way

```
for func in dir(base_class):
    if not callable(getattr(base_class, func)) and func.startswith(
        TestLoader.testMethodPrefix
    ):
        continue
    if func in self._base_class_mapping:
        raise FeathubException(
            f"Duplicated test case name {func} detected in "
            f"integration test base class "
            f"{self._base_class_mapping[func]} and {base_class}."
        )
```